packages/lesswrong/server/collections/advisorRequests/mutations.ts

```diff
@@ -0,0 +1,150 @@
+  // the current action based on whether they own the
+  // document or not.  Otherwise, check if user is an
+  // admin.
+  return userOwns(user, document as HasUserIdType)
+    ? userCanDo(user, [
+      'advisorrequest.update.own',
+      'advisorrequests.edit.own',
+    ])
+    : userCanDo(user, [
+      'advisorrequest.update.all',
+      'advisorrequests.edit.all',
+    ]);
+}
+
+
+const { createFunction, updateFunction } = getDefaultMutationFunctions('AdvisorRequests', {
+  createFunction: async ({ data }: CreateAdvisorRequestInput, context, skipValidation?: boolean) => {
+    const { currentUser } = context;
+
+    const callbackProps = await checkCreatePermissionsAndReturnProps('AdvisorRequests', {
+      context,
+      data,
+      schema,
+      skipValidation,
+    });
+
+    data = callbackProps.document;
+
+    data = await runFieldOnCreateCallbacks(schema, data, callbackProps);
+
+    const afterCreateProperties = await insertAndReturnCreateAfterProps(data, 'AdvisorRequests', callbackProps);
+    let documentWithId = afterCreateProperties.document;
+
+    await runCountOfReferenceCallbacks({
+      collectionName: 'AdvisorRequests',
+      newDocument: documentWithId,
+      callbackStage: 'createAfter',
+      afterCreateProperties,
+    });
+
+    return documentWithId;
+  },
+
+  updateFunction: async ({ selector, data }: UpdateAdvisorRequestInput, context, skipValidation?: boolean) => {
+    const { currentUser, AdvisorRequests } = context;
+
+    // Save the original mutation (before callbacks add more changes to it) for
+    // logging in FieldChanges
+    const origData = cloneDeep(data);
+
+    const {
+      documentSelector: advisorrequestSelector,
+      previewDocument, 
+      updateCallbackProperties,
+    } = await checkUpdatePermissionsAndReturnProps('AdvisorRequests', { selector, context, data, schema, skipValidation });
+
+    const { oldDocument } = updateCallbackProperties;
+
+    const dataAsModifier = dataToModifier(clone(data));
+    data = await runFieldOnUpdateCallbacks(schema, data, dataAsModifier, updateCallbackProperties);
+
+    let modifier = dataToModifier(data);
+
+    // This cast technically isn't safe but it's implicitly been there since the original updateMutator logic
+    // The only difference could be in the case where there's no update (due to an empty modifier) and
+    // we're left with the previewDocument, which could have EditableFieldInsertion values for its editable fields
+    let updatedDocument = await updateAndReturnDocument(modifier, AdvisorRequests, advisorrequestSelector, context) ?? previewDocument as DbAdvisorRequest;
+
+    await runCountOfReferenceCallbacks({
+      collectionName: 'AdvisorRequests',
+      newDocument: updatedDocument,
+      callbackStage: "updateAfter",
+      updateAfterProperties: updateCallbackProperties,
+    });
+
+    void logFieldChanges({ currentUser, collection: AdvisorRequests, oldDocument, data: origData });
```

Comment:
This (and the clone operation at the top) should be in `getDefaultMutationFunctions` (perhaps making `getDefaultMutationFunctions` take a flag that controls whether the changes are logged). If it was one line I'd say ok, w/e, but having it be two nonconsecutive bits (one of which needs a duplicated comment) is too much.